crates/edr_provider/src/requests/serde.rs

```diff
@@ -347,6 +347,63 @@ where
     Ok(result)
 }
 
+/// Value representing timestamp in seconds.
+///
+/// While technically the Yellow Paper allows specifying the timestamp as
+/// any scalar value, using `u64` in practice is more than enough.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize)]
+pub struct Timestamp(pub u64);
+
+impl From<u64> for Timestamp {
+    fn from(value: u64) -> Self {
+        Self(value)
+    }
+}
+
+impl From<Timestamp> for u64 {
+    fn from(value: Timestamp) -> Self {
+        value.0
+    }
+}
+
+impl Timestamp {
+    pub fn as_u64(&self) -> u64 {
```

Comment:
It's not used but provided it for clarity; can remove it if you'd like per YAGNI